specs/Print.md

```diff
@@ -0,0 +1,816 @@
+  catch (NotImplementedException exception)
+  {
+    MessageBox.Show(this, "Printing is failed: " + exception.Message, "Print to default printer");
+  }
+}
+```
+
+## Print to printer
+
+You can use `Print` method to print to a specific printer with the print settings.
+You can configure print settings using `CreatePrintSettings`, and call `Print`
+to print the current web page.
+
+```cpp
+// This example prints the current web page to a specific printer with the settings.
+
+struct PrintSettings
+{
+  COREWEBVIEW2_PRINT_ORIENTATION Orientation = COREWEBVIEW2_PRINT_ORIENTATION_PORTRAIT;
+  int Copies = 1;
+  int PagesPerSheet = 1;
+  std::wstring Pages = L"";
+  COREWEBVIEW2_PRINT_COLLATION Collation = COREWEBVIEW2_PRINT_COLLATION_DEFAULT;
+  COREWEBVIEW2_PRINT_COLOR_MODE ColorMode = COREWEBVIEW2_PRINT_COLOR_MODE_DEFAULT;
+  COREWEBVIEW2_PRINT_DUPLEX Duplex = COREWEBVIEW2_PRINT_DUPLEX_DEFAULT;
+  COREWEBVIEW2_PRINT_MEDIA_SIZE Media = COREWEBVIEW2_PRINT_MEDIA_SIZE_DEFAULT;
+  double PaperWidth = 8.5;
+  double PaperHeight = 11;
+  double ScaleFactor = 1.0;
+  bool PrintBackgrounds = false;
+  bool HeaderAndFooter = false;
+  bool ShouldPrintSelectionOnly = false;
+  std::wstring HeaderTitle = L"";
+  std::wstring FooterUri = L"";
+};
+
+// Function to get printer name by displaying printer text input dialog to the user.
+// User has to specify the desired printer name by querying the installed printers list on the OS
+// to print the web page.
+// You may also choose to display printers list to the user and return user selected printer.
+std::wstring AppWindow::GetPrinterName()
+{
+  std::wstring printerName;
+
+  TextInputDialog dialog(
+    GetMainWindow(), L"Printer Name", L"Printer Name:",
+    L"Specify a printer name from the installed printers list on the OS.", L"");
+
+  if (dialog.confirmed)
+  {
+    printerName = (dialog.input).c_str();
+  }
+  return printerName;
+
+  // or
+  //
+  // Use win32 EnumPrinters function to get locally installed printers.
+  // Display the printer list to the user and get the user desired printer to print.
+  // Return the user selected printer name.
+}
+
+// Function to get print settings for the selected printer.
+// You may also choose get the capabilties from the native printer API, display to the user to get
+// the print settings for the current web page and for the selected printer.
+PrintSettings AppWindow::GetSelectedPrinterPrintSettings(std::wstring printerName)
+{
+  PrintSettings printSettings;
+  printSettings.PrintBackgrounds = true;
+  printSettings.HeaderAndFooter = true;
+
+  return printSettings;
+
+  // or
+  //
+  // Use win32 DeviceCapabilitiesA function to get the capabilities of the selected printer.
+  // Display the printer capabilities to the user along with the page settings.
+  // Return the user selected settings.
+}
+
+bool AppWindow::PrintToPrinter()
+{
+  std::wstring printerName = GetPrinterName();
+  PrintSettings printInfo = GetSelectedPrinterPrintSettings(printerName);
```

Comment:
Variable name should be `printSettings` since the type is `PrintSettings`.

Reply:
Will do

Reply:
Can we also rename the app's type to SamplePrintSettings (and var name samplePrintSettings) to make it clear this isn't the WebView2's print settings type.